tools/dumpobj.ml

```diff
@@ -111,7 +111,7 @@ let rec print_struct_const = function
 (* Print an obj *)
 
 let same_custom x y =
-  Obj.field x 0 = Obj.field (Obj.repr y) 0
+  Obj.field x 0 == Obj.field (Obj.repr y) 0
```

Comment:
This may be correct, but it is a bit dangerous because you need to be a compiler expert to ensure that the code does not load naked pointers as temporary values in an incorrect way. Instead you should use the new-ish `Obj.raw_field` API, which properly boxes the naked/raw data into a `nativeint`. So I would try

```
  Obj.raw_field x 0 = Obj.raw_field (Obj.repr y) 0
```

Reply:
Ah yes, that's much better. Done in the new version.